src/authenticator/formAuthenticator.js

```diff
@@ -86,10 +86,37 @@ FormAuthenticator.prototype.get = function(url){
   }
 
   this._getField(this.passFieldSelector).sendKeys(this.pass);
-  this._getField(this.logonButtonSelector).click().then(function () {
-    // wait for all login actions to complete
-    that.statisticsCollector.authDone();
-  });
+  this._getField(this.logonButtonSelector).click();
+
+  function _waitForAuthorizationButtonEnabled() {
+    return that._getField(that.authorizationButtonSelector).isEnabled().then(function (enabled) {
```

Comment:
isEnabled() works differently and there is quite a difference between isEnabled,isAvailable,isVisible and the check for element existence. That's why we have the helpers like _waitForField. Please use it, no need to implement unique stuff.

Reply:
Yes and this is the main problem because the `Authorize` button does not appear every time and if we use `_waitForField` it will wait for it and then timeout which is not what we wanted.
the other thing is that when it appears, for maybe a second the button is **disabled** (gray-out), so this is the reason for the `isEnabled` check.
The idea is to wait until the button is enabled and then press it.
If the button do not appear the test should continue - this is the common use-case.

Reply:
This sounds reasonable but I am worried of ElementNotFound errors thrown if the element does not appear ?

Reply:
As I have tested it the `catch` that follows the promise handles it gracefully both if it does not exist (it catches the error and exit the wait).  If the button is visible, but not enabled it will wait until it becomes available or timeout.